virtualizarr/manifests/store.py

```diff
@@ -137,34 +140,32 @@ def parse_manifest_index(key: str, chunk_key_encoding: str = ".") -> tuple[int,
     return tuple(int(ind) for ind in parts[1].split(chunk_key_encoding))
 
 
-def find_matching_store(stores: StoreDict, request_key: str) -> StoreRequest:
-    """
-    Find the matching store based on the store keys and the beginning of the URI strings,
-    to fetch data from the appropriately configured ObjectStore.
+def _default_object_store(
+    filepath: str, config: Optional[S3Config] = {}
+) -> tuple[str, ObjectStore]:
+    import obstore as obs
+
+    parsed = urlparse(filepath)
+    if parsed.scheme == "s3":
+        if not config:
+            config["skip_signature"] = True
+        config["virtual_hosted_style_request"] = True
+        config["client_options"] = {"allow_http": True}
+        config["virtual_hosted_style_request"] = False
+        bucket = parsed.netloc
+        return f"s3://{bucket}", obs.store.S3Store(
```

Comment:
This is failing on this [test](https://github.com/zarr-developers/VirtualiZarr/blob/develop/virtualizarr/tests/test_backend.py#L222).  We'll likely need some additional logic to infer bucket region https://developmentseed.org/obstore/latest/troubleshooting/aws/#region-required

Reply:
addressed in [`ca48b6c` (#549)](https://github.com/zarr-developers/VirtualiZarr/pull/549/commits/ca48b6c78b709246ed646c7938780751b109e0dc), thanks for the pointer @sharkinsspatial and the amazing docs @kylebarron!